From f34afa8438421f64dca53af2184a43cd5825dd4f Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Wed, 16 Sep 2026 09:58:35 +0200
Subject: [PATCH 8/9] TEST: Fix the eos-io-tool build on macOS

eos-io-tool is built unconditionally, including for a CLIENT=1 build, and
opens files with O_LARGEFILE, which Darwin does not define. The same
`#define O_LARGEFILE 0` shim is already used in fst/XrdFstOssFile.cc, but
that one is file-local and the file is server-only.
---
 test/eos_io_tool.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/eos_io_tool.cc b/test/eos_io_tool.cc
index 851cb9bed..fc9bdbbe2 100644
--- a/test/eos_io_tool.cc
+++ b/test/eos_io_tool.cc
@@ -47,6 +47,10 @@
 #include "common/Logging.hh"
 #include "common/Mapping.hh"
 
+#ifdef __APPLE__
+#define O_LARGEFILE 0
+#endif
+
 //! Type of operations supported
 enum OperationType {
   RD_SEQU,
